include/wasmtime.hh

```diff
@@ -2263,15 +2377,30 @@ public:
     return wasmtime_memory_size(cx.ptr, &memory);
   }
 
+#ifdef __cpp_lib_span
   /// Returns a `span` of where this memory is located in the host.
   ///
   /// Note that embedders need to be very careful in their usage of the returned
   /// `span`. It can be invalidated with calls to `grow` and/or calls into
   /// WebAssembly.
   std::span<uint8_t> data(Store::Context cx) const {
-    auto *base = wasmtime_memory_data(cx.ptr, &memory);
-    auto size = wasmtime_memory_data_size(cx.ptr, &memory);
-    return {base, size};
+    return {data_ptr(cx), data_size(cx)};
+  }
+#endif
+
+  /// Returns the base pointer in the host's memory of where this memory is
+  /// located.
+  ///
+  /// Note that embedders need to be very careful in their usage of the returned
+  /// pointer. It can be invalidated with calls to `grow` and/or calls into
+  /// WebAssembly.
+  uint8_t *data_ptr(Store::Context cx) const {
```

Comment:
You can always use your own `wasmtime::Span`, because it has implicit conversion to std::span
This method is not needed

Reply:
I was originally hoping to avoid defining a custom `Span` type but the `Func` constructor stopped me. I agree though that if we define a `Span` type may as well use it!